docs/K8SCollectors.md

```diff
@@ -0,0 +1,179 @@
+# **K8S data collectors**
+The Splunk Operator for K8S deploys Splunk Enterprise custom resources across a single/multiple namespace/s. The helper scripts `k8s-splunk-collector.sh` and `k8s-systeminfo-collector.sh` in the [tools directory](https://github.com/splunk/splunk-operator/tree/master/tools/k8s_collectors), collect data from a K8S cluster which runs the Splunk Operator for K8S.
+
+## **Splunk Collector for K8S - k8s-splunk-collector.sh**
+The Splunk collector for K8S collects data from multiple <kubectl> get/describe commands, container logs and diags from the Splunk instances(if opted for) in the context of the current namespace.
+
+The script:
+- Collects [extensive data](#target-folder-breakdown) on your K8S cluster. If there is any data you'd like to keep private please avoid using the script(modify it as per your needs)
+- If diags is opted for, generates Splunk diag on all of the Splunk Instances running inside of Splunk Enterprise CR pods deployed by the Splunk Operator for K8S. The diag generated by the script on the Splunk instance gets deleted after extraction.
+
+If any of the above is not desired, please resort to collecting data manually.
+
+### **Requirements to run the script**
+- Kubeconfig context set to the cluster/namespace running the Splunk Operator for K8S
+- Access to kubectl commands to get data
+- Root access on the host file system to create/delete directories(atleast within the directory where you want to run this script and store logs)
+- Enough space to collect data in a target folder
+
+### **Script run instructions**
+- Run the script using the following command - `sh k8s-splunk-collector.sh -d <flag_to_collect_splunk_diags> -t <target_folder> -l <flag_to_limit_output_by_avoid_kubectl_describe> -s <collect_secret_data>`. There are three options which are configurable through the script:
+    - The `-d` option is used to specify whether Splunk diags needs to be collected. Splunk diags are not collected by default. Set to `true` if Splunk diags are to be collected.
+    - The `-t` option is used to specify a `<target folder>`. This option is `not mandatory`. The script allows you to store the data collected in two different ways:
+        - If the `-t` option is not used, a timestamped folder `tmp-<timestamp>` is created in the present working directory where the data will be written to. 
+            ```
+            Eg. sh k8s-splunk-collector.sh -d true
+            ```                
+        - If the `-t` option is used with valid full path, a timestamped folder `tmp-<timestamp>` is created inside the full path where the data will be written to. Note: If the folder provided doesn't exist, it is created provided the preceeding path exists i.e in the example below if `/Users/randomuser` is present, `/collect` is created. But if the preceeding path `/Users/randomuser` doesn't exist the script runs into unexpected behavior.
```

Comment:
"the script runs into unexpected behavior" -- can you say what happens? Does the script fail and exit? Continue but not create the folder and use a default folder instead?